mempool/reactor.go

```diff
@@ -23,17 +24,27 @@ const (
 )
 
 // MempoolReactor handles mempool tx broadcasting amongst peers.
+// It maintains a map from peer ID to counter, to prevent gossiping txs to the
+// peers you received it from.
 type MempoolReactor struct {
 	p2p.BaseReactor
 	config  *cfg.MempoolConfig
 	Mempool *Mempool
+
+	idMtx   *sync.RWMutex
+	nextID  uint16 // assumes that a node will never have over 65536 peers
```

Comment:
`type peerID uint16`?

Reply:
Actually, I'm not sure that we should make a custom type for this if we go through with https://github.com/tendermint/tendermint/pull/2778#discussion_r232207251

Reply:
I believe the same desire is achieved by having TxInfo?